### ISSUE TITLE
Time to retire USE_DATE_TIME_PRE_1_33_FACET_IO

### ISSUE THREAD

**Issue by jeking3** [boostorg/date_time#61]
1.33.0 came out a long time ago.  Time to retire the pre-1.33.0 code paths.

**Comment by eldiener** [comment]
I think you should query the developers mailing list regarding this potential removal.

**Comment by jeking3** [comment]
There was no feedback indicating this should not be done.

**Comment by eldiener** [comment]
I will remove it and update the 'develop' branch after I do.

**Comment by eldiener** [comment]
It looks to me that all sorts of tests are only being run when USE_DATE_TIME_PRE_1_33_FACET_IO is being defined. These includes tests for:

- date_time_wide_streaming
- date_time_pre_133_facet
- date_time_serialization

I do not see the same tests being run without USE_DATE_TIME_PRE_1_33_FACET_IO being defined. I feel that I am missing something in the bjam file for the tests but I can not see what it is.

This suggests to me that we can not remove the code which contains USE_DATE_TIME_PRE_1_33_FACET_IO.

**Comment by jeking3** [comment]
Hmm, interesting.  Well I suggest we leave this here for now and revisit it once the trac backlog is drained a little more, or someone can pick it up. 

It would seem that date_time_pre_133_facet would be unnecessary entirely based on the name alone.. not sure about the others.

**Comment by jeking3** [comment]
It's possible that serialization integration only happens pre-1.33 because of the streaming and traits introduced in 1.33... While enabling UBSAN builds I found I had to disable the serialization tests which only run in pre-1.33 compatibility mode.  Might be more reason to remove that legacy code...

**Comment by eldiener** [comment]
If serialization needs the pre-1.33 code I do not see how we can remove that code without saying that serialization does not work for iostreams.

**Comment by eldiener** [comment]
Sorry I meant to say:

If serialization needs the pre-1.33 code I do not see how we can remove that code without saying that serialization does not work for date_time.

**Comment by jeking3** [comment]
True, will come back to it eventually and see.

**Comment by JeffGarland** [comment]
So I'm planning to remove support for this in boost 1.73 -- the time has seriously come.  I've made an initial commit that removes the macros the force this mode for long dropped compilers.  I posted a warning on both the mailing lists: crickets.

https://lists.boost.org/boost-users/2020/03/90303.php
https://lists.boost.org/Archives/boost/2020/03/248334.php

I have a 'dt_nolib' branch which I'm going to put onto develop shortly -- removing the library build.  That changes the testing quite dramatically and as part of that I've dropped all the pre-1.33 tests.  The code is not removed, yet -- but that will be soon.
